arangod/IResearch/IResearchFeature.cpp

```diff
@@ -248,33 +263,28 @@ bool iresearchViewUpgradeVersion0_1(TRI_vocbase_t& vocbase,
 
     irs::utf8_path dataPath;
 
-    if (arangodb::ServerState::instance()->isSingleServer() ||
-        arangodb::ServerState::instance()->isDBServer()) {
-      auto* dbPathFeature =
-          arangodb::application_features::ApplicationServer::lookupFeature<arangodb::DatabasePathFeature>(
-              "DatabasePath");
-
-      if (!dbPathFeature) {
-        LOG_TOPIC("67c7e", WARN, arangodb::iresearch::TOPIC)
-            << "failure to find feature 'DatabasePath' while upgrading "
-               "IResearchView from version 0 to version 1";
+    auto* dbPathFeature = ApplicationServer::lookupFeature<arangodb::DatabasePathFeature>("DatabasePath");
 
-        return false;  // required feature is missing
-      }
+    if (!dbPathFeature) {
+      LOG_TOPIC("67c7e", WARN, arangodb::iresearch::TOPIC)
+          << "failure to find feature 'DatabasePath' while upgrading "
+             "IResearchView from version 0 to version 1";
 
-      // original algorithm for computing data-store path
-      static const std::string subPath("databases");
-      static const std::string dbPath("database-");
-
-      dataPath = irs::utf8_path(dbPathFeature->directory());
-      dataPath /= subPath;
-      dataPath /= dbPath;
-      dataPath += std::to_string(vocbase.id());
-      dataPath /= arangodb::iresearch::DATA_SOURCE_TYPE.name();
-      dataPath += "-";
-      dataPath += std::to_string(view->id());
+      return false;  // required feature is missing
     }
 
+    // original algorithm for computing data-store path
+    static const std::string subPath("databases");
+    static const std::string dbPath("database-");
+
+    dataPath = irs::utf8_path(dbPathFeature->directory());
+    dataPath /= subPath;
+    dataPath /= dbPath;
+    dataPath += std::to_string(vocbase.id());
+    dataPath /= arangodb::iresearch::DATA_SOURCE_TYPE.name();
+    dataPath += "-";
+    dataPath += std::to_string(view->id());
+
```

Comment:
is it guaranteed that these paths exist? I think the `databases` subdirectory is completely optional, at least for RocksDB I don't think it will be there by default.

Reply:
that is how arangosearch did it in 3.4